pkg/response/formatter.go

```diff
@@ -0,0 +1,211 @@
+package response
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"gopkg.in/yaml.v3"
+
+	"github.com/leaktk/scanner/pkg/config"
+	"github.com/leaktk/scanner/pkg/logger"
+)
+
+// OutputFormat is the code(int) for each format
+type OutputFormat int
+
+const (
+	// JSON displays the output in JSON format
+	JSON OutputFormat = iota
+	// HUMAN displays the outut in a way that's nice for humans to read
+	HUMAN
+	// TOML displays the output in TOML format
+	TOML
+	// YAML displays the output in YAML format
+	YAML
+	// CSV displays the
+	CSV
+)
+
+type Formatter struct {
+	format OutputFormat
+}
+
+// NewFormatter creates new formatter
+func NewFormatter(cfg config.Formatter) (*Formatter, error) {
+	format, err := GetOutputFormat(cfg.Format)
+	if err != nil {
+		return nil, err
+	}
+	return &Formatter{format: format}, nil
+}
+
+// GetOutputFormat takes the string and returns OutputFormat or an error
+func GetOutputFormat(format string) (OutputFormat, error) {
+	format = strings.ToUpper(format)
+	switch format {
+	case "JSON":
+		return JSON, nil
+	case "HUMAN":
+		return HUMAN, nil
+	case "TOML":
+		return TOML, nil
+	case "YAML":
+		return YAML, nil
+	case "CSV":
+		return CSV, nil
+	default:
+		return JSON, fmt.Errorf("invalid output format option: format=%q", format)
+	}
+}
+
+// SetFormat sets the format using the OutputFormat value
+func (f *Formatter) SetFormat(format OutputFormat) {
+	f.format = format
+}
+
+// Format renders a response structure to the set format as a string
+func (f *Formatter) Format(r *Response) string {
+	var output string
+	switch f.format {
+	case JSON:
+		output = formatJson(r)
+	case HUMAN:
+		output = formatHuman(r)
+	case TOML:
+		output = formatToml(r)
+	case YAML:
+		output = formatYaml(r)
+	case CSV:
+		output = formatCsv(r)
+	}
+	return output
+}
+
+func formatJson(r *Response) string {
+	out, err := json.Marshal(r)
+	if err != nil {
+		logger.Error("could not marshal response: error=%q", err)
+	}
+	return string(out)
+}
+
+func formatHuman(r *Response) string {
+	headers := flattenedResponseFields()
+	for i, header := range headers {
+		// Append a : to each label. Do it once here instead of every loop
+		headers[i] = header + ":"
+	}
+	flat := flattenedResponse(r)
+	var out []string
+	for _, response := range flat {
+		for i, entry := range response {
+			// Specifies width of 26 characters for labels
+			out = append(out, fmt.Sprintf("%-26s%s", headers[i], entry))
+		}
+		out = append(out, "\n")
+	}
+	return strings.Join(out, "\n")
```

Comment:
Thoughts on using a strings builder?

```suggestion
	var out strings.Builder

	headers := flattenedResponseFields()
	flat := flattenedResponse(r)

	for _, response := range flat {
		for i, entry := range response {
			// Specifies width of 26 characters for labels
			fmt.Fprintf(&out, "%-26s: %s\n", headers[i], entry)
		}
		out.WriteRune('\n')
	}

	return out.String()
```

The output would look like this:

```
...
RESULT.ENTROPY            : 2.584963
RESULT.DATE               : 2024-08-14T18:24:03Z
RESULT.NOTES              : message: Add a yarn file
RESULT.RULE.ID            : QqS4RvI6Zmg
RESULT.RULE.DESCRIPTION   : Authorization Header
RESULT.RULE.TAGS          : "type:secret, group:leaktk-testing"
RESULT.CONTACT            : Braxton Plaxco <bplaxco@redhat.com>
RESULT.LOCATION.VERSION   : 0795d84ec8b38c6fe95cb2f0eba52525585c5486
RESULT.LOCATION.PATH      : yarn-4.2.2.cjs
RESULT.LOCATION.RANGE     : L696C4824-L696C4844

...
```